xgcm/test/test_grid.py

```diff
@@ -782,3 +782,14 @@ def test_boundary_kwarg_same_as_grid_constructor_kwarg():
     actual2 = grid2.interp(ds.data_g, ("X", "Y"))
 
     xr.testing.assert_identical(actual1, actual2)
+
+def test_axisdirectionality():
```

Comment:
I think you can probably test this over more of the examples by using:
```python
def test_axisdirectionality(all_datasets):    
    ds, _,_ = all_datasets
    ...
```

This might expose some more edge cases.